docs/storage.md

```diff
@@ -0,0 +1,97 @@
+./data/01BKGTZQ1SYQJTR4PB43C8PD98/chunks/000001
+./data/01BKGTZQ1SYQJTR4PB43C8PD98/tombstones
+./data/01BKGTZQ1HHWHV8FBJXW1Y3W0K
+./data/01BKGTZQ1HHWHV8FBJXW1Y3W0K/meta.json
+./data/01BKGTZQ1HHWHV8FBJXW1Y3W0K/wal
+./data/01BKGTZQ1HHWHV8FBJXW1Y3W0K/wal/000001
+./data/01BKGV7JC0RY8A6MACW02A2PJD
+./data/01BKGV7JC0RY8A6MACW02A2PJD/meta.json
+./data/01BKGV7JC0RY8A6MACW02A2PJD/index
+./data/01BKGV7JC0RY8A6MACW02A2PJD/chunks
+./data/01BKGV7JC0RY8A6MACW02A2PJD/chunks/000001
+./data/01BKGV7JC0RY8A6MACW02A2PJD/tombstones
+```
+
+The initial two-hour blocks are eventually compacted into longer blocks in the background.
+
+Note that a limitation of the local storage is that it is not clustered or replicated. Thus, it is not arbitrarily scalable or durable in the face of disk or node outages and should thus be treated as more of an ephemeral sliding window of recent data. However, if your durability requirements are not strict, you may still succeed in storing up to years of data in the local storage.
+
+## Operational aspects
+
+Prometheus has several flags that allow configuring the local storage. The most important ones are:
+
+* `--storage.tsdb.path`: This determines where Prometheus writes its database. Defaults to `data/`.
+* `--storage.tsdb.retention`: This determines when to remove old data. Defaults to `15d`.
+* `--storage.tsdb.min-block-duration`: This determines the minimum block duration. Defaults to `2h`.
+
+Typically you will only need to tune the first two, depending on where you want the storage to write its data, and for how long you want to keep it.
+
+On average, Prometheus uses only around 1-2 bytes per sample. Thus, to plan the capacity of a Prometheus server, you can use the rough formula:
+
+```
+needed_disk_space = retention_time_seconds * ingested_samples_per_second * bytes_per_sample
+```
+
+To tune the rate of ingested samples per second, you can either reduce the number of time series you scrape (fewer targets or fewer series per target), or you can reduce the scrape rate. However, reducing the number of series is likely more effective.
+
+If your local storage becomes corrupted for whatever reason, your best bet is to shut down Prometheus and remove the entire storage directory. However, you can also try removing individual block directories to resolve the problem. This means losing a time window of around two hours worth of data per block directory. Again, Prometheus's local storage is not meant as a durable long-term storage.
+
+## Remote storage integrations
+
+As outlined above, Prometheus's local storage is limited in its scalability and durability. Instead of trying to solve long-term storage in Prometheus itself, Prometheus has a set of interfaces that allow integrating with remote long-term storage systems.
+
+### Overview
+
+Prometheus integrates with remote storage systems in two ways:
+
+* Prometheus can write all samples that it ingests to a remote URL in a standardized format.
```

Comment:
s/all//

Relabelling allows restricting this.

Reply:
Ack